Sources/Diagnostics/FileHandler.swift

```diff
@@ -0,0 +1,256 @@
+                break
+            }
+
+            // After detecting `count` lines, start writing
+            linesDetected += string.countOccurences(of: Self.lineBreak)
+            if linesDetected >= count {
+                let linesToWrite = string
+                    .components(separatedBy: String(Self.lineBreak))
+                    .suffix(linesDetected - count + 1)
+                    .joined(separator: String(Self.lineBreak))
+
+                outputFile.write(linesToWrite.asData)
+            }
+        } while true
+
+        // Replace with temporary file
+        try self.replaceHandler(with: tempURL)
+    }
+
+    // MARK: -
+
+    private static let fileManager: FileManager = .default
+
+    private static let lineBreak: Character = "\n"
+    private static let lineBreakData = String(FileHandler.lineBreak).asData
+    private static let bufferSize = 4096
+
+}
+
+// MARK: - Errors
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+extension FileHandler {
+
+    enum Error: Swift.Error {
+
+        case failedCreatingFile(URL)
+        case failedCreatingDirectory(URL)
+        case failedCreatingHandle(Swift.Error)
+        case failedSeeking(Swift.Error)
+        case failedEmptyingFile(Swift.Error)
+        case failedMovingNewFile(from: URL, toURL: URL, Swift.Error)
+
+    }
+
+}
+
+// MARK: - Private
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+private extension FileHandler {
+
+    func moveToBeginningOfFile() throws {
+        do {
+            try self.fileHandle.seek(toOffset: 0)
+        } catch {
+            throw Error.failedSeeking(error)
+        }
+    }
+
+    static func createFileIfNecessary(_ url: URL) throws {
+        guard !Self.fileManager.fileExists(atPath: url.path) else { return }
+
+        let directoryURL = url.deletingLastPathComponent()
+        if !Self.fileManager.fileExists(atPath: directoryURL.path) {
+            do {
+                Logger.verbose(Message.creating_directory(directoryURL))
+
+                try Self.fileManager.createDirectory(at: directoryURL,
+                                                     withIntermediateDirectories: true,
+                                                     attributes: nil)
+            } catch {
+                throw Error.failedCreatingDirectory(directoryURL)
+            }
+        }
+
+        Logger.verbose(Message.creating_file(url))
+
+        if !Self.fileManager.createFile(atPath: url.path, contents: nil, attributes: nil) {
+            throw Error.failedCreatingFile(url)
+        }
+    }
+
+    static func createTemporaryFile() throws -> URL {
+        let result = Self.fileManager.temporaryDirectory
+            .appendingPathComponent("com.revenenuecat")
+            .appendingPathComponent(UUID().uuidString)
+
+        try Self.createFileIfNecessary(result)
+        return result
+    }
+
+    func replaceHandler(with otherURL: URL) throws {
```

Comment:
I wonder if this function should be executed atomically, so if there are multiple threads using this handler, this doesn't cause any issues (We don't do this in Android either)

Reply:
That's handled by the fact that this type is an `actor` :) 
https://www.swiftbysundell.com/articles/swift-actors/

Reply:
Ahh right!